src/data/nat/choose/factorization.lean

```diff
@@ -0,0 +1,150 @@
+end
+
+/--
+A `pow` form of `nat.factorization_choose_le`
+-/
+lemma pow_factorization_choose_le (hn : 0 < n) : p ^ (choose n k).factorization p ≤ n :=
+begin
+  cases le_or_lt p 1,
+  { exact (pow_le_pow_of_le h).trans ((le_of_eq (one_pow _)).trans hn) },
+  { exact (pow_le_iff_le_log h hn).mpr factorization_choose_le },
+end
+
+/--
+Primes greater than about `sqrt n` appear only to multiplicity 0 or 1 in the binomial coefficient.
+-/
+lemma factorization_choose_le_one (p_large : n < p ^ 2) : (choose n k).factorization p ≤ 1 :=
+begin
+  apply factorization_choose_le.trans,
+  rw [←not_lt, one_lt_iff_ne_zero_and_ne_one, not_and_distrib, not_not, not_not,
+      log_eq_zero_iff, log_eq_one_iff, ←sq, and_iff_right p_large, and_comm,
+      ←not_lt, ←not_le, ←not_and_distrib, or_comm],
+  exact or_not,
+end
+
+lemma factorization_choose_of_lt_three_mul
+  (hp' : p ≠ 2) (hk : p ≤ k) (hk' : p ≤ n - k) (hn : n < 3 * p) :
+  (choose n k).factorization p = 0 :=
+begin
+  cases em' p.prime with hp hp,
+  { exact factorization_eq_zero_of_non_prime (choose n k) hp },
+  cases lt_or_le n k with hnk hkn,
+  { rw [choose_eq_zero_of_lt hnk, factorization_zero, finsupp.coe_zero, pi.zero_apply] },
+  rw [←@padic_val_nat_eq_factorization p _ ⟨hp⟩, @padic_val_nat_def _ ⟨hp⟩ _ (choose_pos hkn)],
+  simp only [hp.multiplicity_choose hkn (lt_add_one _), enat.get_coe,
+    finset.card_eq_zero, finset.filter_eq_empty_iff, not_le],
+  intros i hi,
+  rcases eq_or_lt_of_le (finset.mem_Ico.mp hi).1 with rfl | hi,
+  { rw [pow_one, ←add_lt_add_iff_left (2 * p), ←succ_mul, two_mul, add_add_add_comm],
+    exact lt_of_le_of_lt (add_le_add
+      (add_le_add_right (le_mul_of_one_le_right' ((one_le_div_iff hp.pos).mpr hk)) (k % p))
+      (add_le_add_right (le_mul_of_one_le_right' ((one_le_div_iff hp.pos).mpr hk')) ((n - k) % p)))
+      (by rwa [div_add_mod, div_add_mod, add_tsub_cancel_of_le hkn]) },
+  { replace hn : n < p ^ i,
+    { calc n < 3 * p : hn
+      ... ≤ p * p : mul_le_mul_right' (lt_of_le_of_ne hp.two_le hp'.symm) p
+      ... = p ^ 2 : (sq p).symm
+      ... ≤ p ^ i : pow_le_pow hp.one_lt.le hi },
+    rwa [mod_eq_of_lt (lt_of_le_of_lt hkn hn), mod_eq_of_lt (lt_of_le_of_lt tsub_le_self hn),
+      add_tsub_cancel_of_le hkn] },
+end
+
+/--
+Primes greater than about `2 * n / 3` and less than `n` do not appear in the factorization of
+`central_binom n`.
+-/
+lemma factorization_central_binom_of_two_mul_self_lt_three_mul
+  (n_big : 2 < n) (p_le_n : p ≤ n) (big : 2 * n < 3 * p) :
+  ((central_binom n).factorization p) = 0 :=
+begin
+  refine factorization_choose_of_lt_three_mul _ p_le_n (p_le_n.trans _) big,
+  { intro h,
+    rw [h, mul_comm, mul_lt_mul_right] at big,
+    all_goals { linarith } },
+  { rw [two_mul, add_tsub_cancel_left] },
+end
+
+lemma factorization_eq_zero_of_lt (h : n < p) : n.factorization p = 0 :=
+finsupp.not_mem_support_iff.mp (mt le_of_mem_factorization (not_le_of_lt h))
```

Comment:
This should go in `data/nat/factorization`, near `factorization_eq_zero_of_non_prime`

Reply:
Done!